src/components/UserPage/UserPage.js

```diff
@@ -1,10 +1,135 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import { connect } from 'react-redux';
 import LogOutButton from '../LogOutButton/LogOutButton';
 import mapStoreToProps from '../../redux/mapStoreToProps';
 
+const mapStateToProps = (state, ownProps) => ({
+  dispatch: ownProps.dispatch,
+  history: ownProps.history,
+  user: state.user,
+  friends: state.friendsReducer,
+  decks: state.deck,
+})
+
+const UserPageFunc = ({
+  dispatch,
+  history,
+  user,
+  friends,
+  decks,
+}) => {
+  useEffect(() => {
+    dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'GET_DECK' });
+    dispatch({ type: 'GET_FRIENDS' });
+  }, [dispatch])
+
+  const createDeck = () => {
+    const deck = prompt("Please enter a deck name:", "My Awesome Deck");
+    if (deck === null || deck === "") {
+      alert("Canceled");
+    } else {
+      this.props.dispatch({
+        type: 'CREATE_DECK',
+        payload: {
+          userid: user.id,
+          deckname: deck,
+          ispublic: false,
+          description: "",
+          decklist: '',
+          featured_card: '',
+          upvotes: 0,
+          comments: '',
+        },
+      });
+    };
+  };
+
+  const deleteDeck = (deck) => {
+    const response = window.confirm(`Are you sure you want to permanenently delete this deck?`)
+    if (response){
+      this.props.dispatch({
+        type: 'DELETE_DECK',
+        payload: deck
+      })
+    }
+  };
+
+  return (
+    <div id='userMainDiv'>
+      <img id='profilePic' height='130px' width='130px' src={user.img_url} />
+      <div id='profileInfo'>
+        <h1 id='welcome'>
+          Welcome, {user.username}!
+        </h1>
+        <button onClick={() => history.push('/edituser')}>
```

Comment:
This is a great example of where we did not need an entire method declared just to navigate to a different page.